### PR TITLE
Feature: Dependency injection

### LINKED ISSUE #63: Stabilise E2E tests
Implement dependency injection to avoid reliance on Firebase for the tests.

### PR DESCRIPTION
**Pull request by matttsss**
### What is this PR?
This PR aims to reduce dependency of tests on Firebase by using Hilt for dependency injections.

### How?
Hilt was added to the project framework and injects repositories into the `MainActivity`.
It currently has three modules:
* `AppModule`: This module provides the `FirebaseFirestore` and `FirebaseAuth` instances with the correct settings
*  `RepoModule`: This is the main module used to provide production repositories
*  `MockRepoModule`: This is the testing module used to provide Mock repositories during testing

This PR also provides Mock implementations for the repositories and can be found in `app/src/androidTest/java/com/github/se/cyrcle/di/mocks`, to be used when necessary in future tests.

### Code Coverage:
image

Closes #63

### CODE REVIEW

**Review by Armetiis — APPROVED** [review]
Thanks chief 🫡

More seriously, amazing work you did here to integrate Dagger Hilt to the project. The mock folder with the provided mock repositories is a great idea.

Also, thank you for aligning the build.gradle.kts!

File: app/src/androidTest/java/com/github/se/cyrcle/di/mocks/MockParkingRepository.kt [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+import com.github.se.cyrcle.model.parking.Parking
+import com.github.se.cyrcle.model.parking.ParkingRepository
+import com.github.se.cyrcle.model.parking.TestInstancesParking
+import com.mapbox.geojson.Point
+import com.mapbox.turf.TurfMeasurement
+import javax.inject.Inject
+
+class MockParkingRepository @Inject constructor() : ParkingRepository {
+  private var uid = 0
+  val parkings = mutableListOf<Parking>(TestInstancesParking.parking1)
```
Comment (sean-prz) [inline app/src/androidTest/java/com/github/se/cyrcle/di/mocks/MockParkingRepository.kt:12]: Everything in the mocks are private expect for this val

**Review by sean-prz — APPROVED** [review]
Wow this is a huge amount of works to understand Hilt and then write explicitly every function of the repository !
Thank you so much, it will make testing way more pleasant.

tests still works on my end, and the app functionality is preserved. We're good to go.

File: app/src/androidTest/java/com/github/se/cyrcle/di/mocks/MockParkingRepository.kt (hunk 1 above)
Comment (matttsss) [reply app/src/androidTest/java/com/github/se/cyrcle/di/mocks/MockParkingRepository.kt:12]: Right, that was for debuging the E2E tests xD

**Comment by sean-prz** [comment]
We will still need to address the fact that we can't add more parking spot to the db without braking the unit test for the parking repository. 

I don't think that Dependency Injection can fix this as we want to test the actual repository, what did you think of Maria's idea of adding the debug variable to the repo to change collection path

**Review by Armetiis — APPROVED** [review]

**Comment by matttsss** [comment]
@sean-prz I do not think that this is a good idea, we would still rely on Firebase and keep on making requests. I realy do not think  that our repo tests should exist in the `androidTest` folder. I beleive that Firestore should be mocked just like it currently is in the `ParkingRepositoryFirestoreTest` from the `test` folder

**Pull request merged**